pkg/asset/machines/aws/machines.go

```diff
@@ -37,19 +52,20 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 		if len(subnets) > 0 && !ok {
 			return nil, nil, errors.Errorf("no subnet for zone %s", zone)
 		}
-		provider, err := provider(
-			clusterID,
-			region,
-			subnet,
-			mpool.InstanceType,
-			&mpool.EC2RootVolume,
-			mpool.EC2Metadata,
-			mpool.AMIID,
-			zone,
-			role,
-			userDataSecret,
-			userTags,
-		)
+		provider, err := provider(&machineProviderInput{
+			clusterID:      clusterID,
+			region:         region,
+			subnet:         subnet,
+			instanceType:   mpool.InstanceType,
+			osImage:        mpool.AMIID,
+			zone:           zone,
+			role:           role,
+			userDataSecret: userDataSecret,
+			root:           &mpool.EC2RootVolume,
+			imds:           mpool.EC2Metadata,
+			userTags:       userTags,
+			publicSubnet:   false,
```

Comment:
Why add `publicSubnet` as an arg to the function if it is just a static value of false? Why not just use the `false` value within the function itself?

Reply:
the `provider()`[1] function is used by `MachineSet`[2] which is not static.
[1] https://github.com/openshift/installer/pull/6371/commits/2e18401e10feb76d4ec25b6088253a4952049614#diff-e46d61c55e5e276e3c264d18cba0346777fe3e662d0180a173001b8282af7c6eR172
[2] https://github.com/openshift/installer/blob/2e18401e10feb76d4ec25b6088253a4952049614/pkg/asset/machines/aws/machinesets.go

https://github.com/openshift/installer/pull/6371/commits/2e18401e10feb76d4ec25b6088253a4952049614

Reply:
`MachineSets()` are responsible to generate the worker (and not edge) MachineSets manifests